doc/api/index.rst

```diff
@@ -1,67 +1,49 @@
-API Overview
-============
+API
+===
 
 .. toctree::
    :hidden:
 
    api_changes
 
-.. contents:: :local:
+For recent changes, see :doc:`api_changes`.
 
-See also the :doc:`api_changes`.
+Much of the library is accessed by creating :doc:`Figure <figure_api>` 
+and :doc:`Axes <axes_api>` objects and calling methods on them. 
 
-Usage patterns
---------------
-
-Below we describe several common approaches to plotting with Matplotlib.
-
-The pyplot API
-^^^^^^^^^^^^^^
-
-`matplotlib.pyplot` is a collection of command style functions that make
-Matplotlib work like MATLAB. Each pyplot function makes some change to a
-figure: e.g., creates a figure, creates a plotting area in a figure, plots
-some lines in a plotting area, decorates the plot with labels, etc.
-
-`.pyplot` is mainly intended for interactive plots and simple cases of
-programmatic plot generation.
-
-Further reading:
-
-- The `matplotlib.pyplot` function reference
-- :doc:`/tutorials/introductory/pyplot`
-- :ref:`Pyplot examples <pyplots_examples>`
-
-.. _api-index:
+  - :doc:`figure_api`: axes creation, figure-level artists
+  - :doc:`axes_api`: most plotting methods, axes labels, access to 
+    axis styling, etc.  
 
-The object-oriented API
-^^^^^^^^^^^^^^^^^^^^^^^
+   
+(Note that there is also a parallel `matplotlib.pyplot` API interface 
+that is considered useful for interactive work; see 
+:ref:`usage patterns <usage_patterns>`, below).
 
-At its core, Matplotlib is object-oriented. We recommend directly working
-with the objects, if you need more control and customization of your plots.
+Simple example: the ``ax`` object has an `.axes.Axes.plot` method 
+that is used to create the line plot.   
+ 
+.. plot::
+   :include-source:
+   :align: center
 
-In many cases you will create a `.Figure` and one or more
-`~matplotlib.axes.Axes` using `.pyplot.subplots` and from then on only work
-on these objects. However, it's also possible to create `.Figure`\ s
-explicitly (e.g. when including them in GUI applications).
-
-Further reading:
+   import matplotlib.pyplot as plt
+   import numpy as np
 
-- `matplotlib.axes.Axes` and `matplotlib.figure.Figure` for an overview of
-  plotting functions.
-- Most of the :ref:`examples <examples-index>` use the object-oriented approach
-  (except for the pyplot section)
+   x = np.arange(0, 4, 0.1)
+   y = np.sin(x*np.pi)
 
-The pylab API (disapproved)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   fig, ax = plt.subplots(figsize=(3,2), constrained_layout=True)
```

Comment:
```suggestion
   fig, ax = plt.subplots()
```
Keep it simple without parameters. While `constrained_layout` is nice, IMHO this is not the place to teach it (and the benefit for this simple plot is small).

Reply:
Yes, but then the figure is too large, in my opinion.

Reply:
You could try with

~~~
.. plot::
   :scale: 50 %
~~~

but I'm not sure if that will look good.

Reply:
Yeah, I wish there was a two-column view....  We could just include the code and the figure as an image as well. I do think the example is useful, though, to make it concrete what we are talking about.

Reply:
By default, the ylabel is cutoff, so *some* formatting is necessary anyway..  Might as well keep what is above.  I don't think we need to explain every incantation everywhere it appears.